File: R.Conocimiento/CodigosRepresentacionConocimiento/RepresentacionConocimiento.py
# Representación del conocimiento en Python

# Definición de hechos
hechos = {
    'Juan': {
        'edad': 25,
        'profesion': 'Ingeniero',
        'ciudad': 'Madrid'
    },
    'Maria': {
        'edad': 30,
        'profesion': 'Abogada',
        'ciudad': 'Barcelona'
    }
}

# Definición de reglas
def es_mayor_de_edad(persona):
    if hechos[persona]['edad'] >= 18:
        return True
    else:
        return False

# Consulta de conocimiento
nombre = 'Juan'
print(f"¿{nombre} es mayor de edad?")
if es_mayor_de_edad(nombre):
    print("Sí, es mayor de edad.")
else:
    print("No, no es mayor de edad.")

File: R.Conocimiento/CodigosRepresentacionConocimiento/RepresentacionConocimiento.cpp
#include <iostream>
#include <vector>

// Definición de la estructura de conocimiento
struct Persona {
    std::string nombre;
    std::string ocupacion;
};

int main() {
    // Creación de la base de conocimientos
    std::vector<Persona> conocimiento;

    // Agregar hechos a la base de conocimientos
    conocimiento.push_back({"Juan", "estudiante"});
    conocimiento.push_back({"Maria", "profesora"});
    conocimiento.push_back({"Pedro", "estudiante"});

    // Consultas de conocimiento
    std::string nombreConsulta;
    std::cout << "Ingrese el nombre de una persona: ";
    std::cin >> nombreConsulta;

    bool encontrado = false;

    // Recorrer la base de conocimientos y buscar la ocupación de la persona consultada
    for (const auto& persona : conocimiento) {
        if (persona.nombre == nombreConsulta) {
            std::cout << "La ocupacion de " << persona.nombre << " es " << persona.ocupacion << std::endl;
            encontrado = true;
            break;
        }
    }

    if (!encontrado) {
        std::cout << "No se encontro información sobre la persona consultada." << std::endl;
    }

    return 0;
}
